hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadata.java

```diff
@@ -232,6 +233,18 @@ Map<Pair<String, String>, List<HoodieMetadataColumnStats>> getColumnStats(List<P
    */
   HoodiePairData<String, HoodieRecordGlobalLocation> readRecordIndex(HoodieData<String> recordKeys);
 
+  /**
+   * Returns the location of record keys which are found in the record index.
+   * Records that are not found are ignored and wont be part of map object that is returned.
+   * @param recordKeys list of recordkeys to look up in the index
+   * @param dataTablePartition option of the data table partition to look up from. This is only applicable to partitioned rli
+   *                           record keys are not globally unique so the record key alone does not sufficiently identify an
+   *                           individual record.
+   * @return map from recordkey to the location of the record that was read from the index
+   *
+   */
+  HoodiePairData<String, HoodieRecordGlobalLocation>  readRecordIndex(HoodieData<String> recordKeys, Option<String> dataTablePartition);
```

Comment:
The param `Option<String> dataTablePartition` is confusing, if the partitioned RLI always got a partition param there, we should rename the method to `readLocalRecordIndex` or something and make the param `dataTablePartition ` just as `String.`